lib/ui/detail/view_model/detail_view_model.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+import 'package:weather_app_flutter_mvvm/repository/weather_repository_provider.dart';
+import 'package:weather_app_flutter_mvvm/ui/detail/ui_state/detail_ui_state.dart';
+
+part 'detail_view_model.g.dart';
+
+@riverpod
+class DetailViewModel extends _$DetailViewModel {
+  @override
+  FutureOr<DetailUiState?> build() {
```

Comment:
> 初期で持たせるbuild関数の戻り値でかなり迷いました。。
> できれば初期値は AsyncLoadingにしたかったのですが、うまくいかず。
> 結果的にnullを許容するようにしてbuild関数ではnullを返すようにしました。

うまくいかなかった場合のエラー等が不明なので、気になるところを書きます
この原因は、FutureOr型にしていることが原因かもしれません
FutureOr<T>は、Future＜T>でもTでも良い、みたいな扱いになるようです
Orは不要かと思います

それか、AsyncLoadingに型の定義が必要か、だと思います
（テストの時はそんなことがあったような）

Reply:
FuterOrからFutureに変更しました。 481b4afedc1b36b156b250ff13120ec3ee30e04d

```
@override
  Future<DetailUiState?> build() {
    return AsyncLoading();
  }
```

できればこのようにしてfetchWeatherメソッドが呼ばれるまで初期値はAsyncLoadingにしたかったのですが、
```
A value of type 'AsyncLoading<dynamic>' can't be returned from the method 'build' because it has a return type of 'Future<DetailUiState?>'.dart[return_of_invalid_type]
```

のエラーになります(戻り値をFuture<DetailUiState?>にしているため当たり前かもしれませんが...)

そもそもbulidメソッド内の初期値としてAsyncLoadingを返すというのはできないのですかね..

Reply:
@oggysy 
<>をつけて、その内に型を定義したら通過しませんか？
型を描かない場合、ダイナミック型になっているせいかと思っています〜
AsyncLoading<dynamic>'

Reply:
```
Future<DetailUiState?> build() {
    return AsyncLoading<DetailUiState?>();
  }
```

も試してみましたが以下のエラーになりました。

```
A value of type 'AsyncLoading<DetailUiState?>' can't be returned from the method 'build' because it has a return type of 'Future<DetailUiState?>'.dart[return_of_invalid_type]
```

NotifierProviderの仕組みを理解しきれていない部分もあるのですが、
buildメソッドが返す値に期待される値はAsyncValue型でないためできないと思いました。
(AsyncValue型として操作できるのはstateプロパティとして扱えるようになってから？)

buildメソッドの戻り値をAsyncValue<DetailUiState?>とするのも試しましたが以下のエラーでダメでした。
```
DetailViewModel.build' ('AsyncValue<DetailUiState?> Function()') isn't a valid override of 'AutoDisposeAsyncNotifier.build' ('FutureOr<DetailUiState?> Function()'
```

Reply:
ありがとうございます！
```
AsyncValue<DetailUiState> build() {
    return const AsyncLoading<DetailUiState>();
  }
```
としたらできました。